src/main/java/bisq/core/payment/payload/MoneyGramAccountPayload.java

```diff
@@ -0,0 +1,140 @@
+
+import javax.annotation.Nullable;
+
+@EqualsAndHashCode(callSuper = true)
+@ToString
+@Setter
+@Getter
+@Slf4j
+public class MoneyGramAccountPayload extends CountryBasedPaymentAccountPayload {
+    private String holderName;
+    private String city;
+    private String state = ""; // is optional. we don't use @Nullable because it would makes UI code more complex.
+    private String email;
+
+    public MoneyGramAccountPayload(String paymentMethod, String id) {
+        super(paymentMethod, id);
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // PROTO BUFFER
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    private MoneyGramAccountPayload(String paymentMethodName,
+                                    String id,
+                                    String countryCode,
+                                    String holderName,
+                                    String city,
+                                    String state,
+                                    String email,
+                                    long maxTradePeriod,
+                                    @Nullable Map<String, String> excludeFromJsonDataMap) {
+        super(paymentMethodName,
+                id,
+                countryCode,
+                maxTradePeriod,
+                excludeFromJsonDataMap);
+        this.holderName = holderName;
+        this.city = city;
+        this.state = state;
+        this.email = email;
+    }
+
+    @Override
+    public Message toProtoMessage() {
+        PB.MoneyGramAccountPayload.Builder builder =
+                PB.MoneyGramAccountPayload.newBuilder()
+                        .setHolderName(holderName)
+                        .setCity(city)
+                        .setState(state)
+                        .setEmail(email);
+
+        final PB.CountryBasedPaymentAccountPayload.Builder countryBasedPaymentAccountPayload = getPaymentAccountPayloadBuilder()
+                .getCountryBasedPaymentAccountPayloadBuilder()
+                .setMoneyGramAccountPayload(builder);
+        return getPaymentAccountPayloadBuilder()
+                .setCountryBasedPaymentAccountPayload(countryBasedPaymentAccountPayload)
+                .build();
+    }
+
+    public static PaymentAccountPayload fromProto(PB.PaymentAccountPayload proto) {
+        PB.CountryBasedPaymentAccountPayload countryBasedPaymentAccountPayload = proto.getCountryBasedPaymentAccountPayload();
+        PB.MoneyGramAccountPayload moneyGramAccountPayload = countryBasedPaymentAccountPayload.getMoneyGramAccountPayload();
+        return new MoneyGramAccountPayload(proto.getPaymentMethodId(),
+                proto.getId(),
+                countryBasedPaymentAccountPayload.getCountryCode(),
+                moneyGramAccountPayload.getHolderName(),
+                moneyGramAccountPayload.getCity(),
+                moneyGramAccountPayload.getState(),
```

Comment:
State is optional and need to be treated in PB optional otherwise it can cause errors. Is the same in WesternUnion (my fault).